hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/RecoveredReplicationSource.java

```diff
@@ -104,7 +106,30 @@ public void locateRecoveredPaths(String walGroupId) throws IOException {
         // didn't find a new location
         LOG.error(
           String.format("WAL Path %s doesn't exist and couldn't find its new location", path));
-        newPaths.add(path);
+        Path walPath = path;
+        if (
```

Comment:
So for RecoveredReplicationSource, we could get a Path under the oldWALs directory? This is by design?

Reply:
After we create RecoveredReplicationSource, we will add a list of WAL files, and the Path under the oldWALs directory.
```
// ReplicationSourceManager.java:700
    oldsources.add(src);
      LOG.info("Added source for recovered queue {}", src.getQueueId());
      for (String wal : walsSet) {
        LOG.trace("Enqueueing log from recovered queue for source: " + src.getQueueId());
        src.enqueueLog(new Path(oldLogDir, wal));
      }
      src.startup();
```
I tried to change it to the actual path, but if the queue has been  failover 2 or more, it will require a lot of judgments and  parse.

The current version of the code tries to put the original generated path。

Reply:
When we restore, the Path from Zookeeper(or other storage). The Zookeeper path does not contain the filesystem path. Judging from the code, it might be to reduce the complexity of modifying Zookeeper when moving files to oldWALs.